### PR TITLE
FontEditor: Random improvements

### LINKED ISSUE #10900: FontEditor: Doesn't allow to distinguish emoji from normal glyphs
https://user-images.githubusercontent.com/42967349/141642775-2657a712-99c3-4419-9084-c6889b0500b3.mp4

### PR DESCRIPTION
**Pull request by sppmacd**
**FontEditor: Display something for emoji glyphs**

This commit makes FontEditor displaying emojis in GlyphMapWidget. They
are not editable, what is marked by red background of a glyph.

Additionally, a proper information is displayed in statusbar.

Fixes #10900.

**FontEditor: Save scale to config file**
image

### CODE REVIEW

**Comment by linusg** [comment]
Why "(somehow)"?

**Comment by sppmacd** [comment]
I don't know

File: Userland/Applications/FontEditor/GlyphEditorWidget.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include "GlyphEditorWidget.h"
 #include <AK/StringBuilder.h>
 #include <AK/UnicodeUtils.h>
+#include <LibConfig/Client.h>
```
Comment (linusg) [inline Userland/Applications/FontEditor/GlyphEditorWidget.cpp:10]: Why is this being included here?

File: Userland/Applications/FontEditor/FontEditor.h [hunk 2]
```diff
@@ -15,7 +16,8 @@
 class GlyphEditorWidget;
 class GlyphMapWidget;
 
-class FontEditorWidget final : public GUI::Widget {
+class FontEditorWidget final : public GUI::Widget
+    , public Config::Listener {
```
Comment (linusg) [inline Userland/Applications/FontEditor/FontEditor.h:20]: Please format this as

```suggestion
class FontEditorWidget final
    : public GUI::Widget
    , public Config::Listener {
```

File: Userland/Applications/FontEditor/GlyphEditorWidget.cpp (hunk 1 above)
Comment (sppmacd) [reply Userland/Applications/FontEditor/GlyphEditorWidget.cpp]: Fixed.

File: Userland/Applications/FontEditor/FontEditor.h (hunk 2 above)
Comment (sppmacd) [reply Userland/Applications/FontEditor/FontEditor.h]: Fixed.

File: Userland/Applications/FontEditor/FontEditor.cpp [hunk 3]
```diff
@@ -293,26 +295,26 @@ FontEditorWidget::FontEditorWidget(const String& path, RefPtr<Gfx::BitmapFont>&&
...
-    m_scale_five_action = GUI::Action::create_checkable("500%", { Mod_Ctrl, Key_1 }, [&](auto&) {
-        m_glyph_editor_widget->set_scale(5);
-        m_glyph_editor_container->set_fixed_size(m_glyph_editor_widget->preferred_width(), m_glyph_editor_widget->preferred_height());
-        m_left_column_container->set_fixed_width(m_glyph_editor_widget->preferred_width());
+    i32 scale = Config::read_i32("FontEditor", "GlyphEditor", "Scale", 10);
+    m_glyph_editor_widget->set_scale(scale);
+
+    m_scale_five_action = GUI::Action::create_checkable("500%", { Mod_Ctrl, Key_1 }, [this](auto&) {
+        set_scale(5);
+        Config::write_i32("FontEditor", "GlyphEditor", "Scale", 5);
```
Comment (Undefine-Unity) [inline Userland/Applications/FontEditor/FontEditor.cpp:303]: Maybe you should move this to the set_scale function, same with the saving below.

File: Userland/Applications/FontEditor/FontEditor.cpp [hunk 4]
```diff
@@ -293,26 +295,26 @@ FontEditorWidget::FontEditorWidget(const String& path, RefPtr<Gfx::BitmapFont>&&
...
     });
-    m_scale_ten_action->set_checked(true);
+    m_scale_ten_action->set_checked(scale == 10);
     m_scale_ten_action->set_status_tip("Scale the editor in proportion to the current font");
-    m_scale_fifteen_action = GUI::Action::create_checkable("1500%", { Mod_Ctrl, Key_3 }, [&](auto&) {
-        m_glyph_editor_widget->set_scale(15);
-        m_glyph_editor_container->set_fixed_size(m_glyph_editor_widget->preferred_width(), m_glyph_editor_widget->preferred_height());
-        m_left_column_container->set_fixed_width(m_glyph_editor_widget->preferred_width());
+    m_scale_fifteen_action = GUI::Action::create_checkable("1500%", { Mod_Ctrl, Key_3 }, [this](auto&) {
+        set_scale(10);
```
Comment (Undefine-Unity) [inline Userland/Applications/FontEditor/FontEditor.cpp:314]: Why is it 10, shouldn't it be 15?

Comment (sppmacd) [reply Userland/Applications/FontEditor/FontEditor.cpp]: Fixed

File: Userland/Applications/FontEditor/FontEditor.cpp (hunk 3 above)
Comment (sppmacd) [reply Userland/Applications/FontEditor/FontEditor.cpp]: Fixed

**Comment by bgianfo** [comment]
@sppmacd  can you fix the merge conflicts?

**Comment by sppmacd** [comment]
Done

**Comment by sppmacd** [comment]
This was merged with #11621.

**Pull request closed without merging**